manager/container.go

```diff
@@ -270,6 +273,44 @@ func (c *containerData) updateLoad(newLoad uint64) {
 	}
 }
 
+func (c *containerData) updateSubcontainerStats(stats *info.ContainerStats, path string) (*info.ContainerStats, error) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	for _, sub := range c.contSubcontainers {
+		subLoadStats, err := sub.getLoadStats()
+		if err != nil {
+			glog.Errorf("could not get cpuload for container %v. got error %v", sub.info.Name, err)
+			continue
+		}
+		glog.V(3).Infof("c: %v, sub %v, subLoadStats running: %+v", c.info.Name, sub.info.Name, subLoadStats.NrRunning)
+		stats.UpdateTaskStats(subLoadStats)
+		c.updateLoad(subLoadStats.NrRunning)
+		// convert to 'milliLoad' to avoid floats and preserve precision.
+		stats.Cpu.LoadAverage = int32(c.loadAvg * 1000)
+	}
+	return stats, nil
+}
+
+func (c *containerData) getLoadStats() (info.LoadStats, error) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	var loadstats info.LoadStats
+	if time.Since(c.lastCpuLoadGrab) > 2*time.Second {
```

Comment:
why 2 seconds? Our normal housekeeping is 1 second. so this will always be true.